### PR TITLE
adds instance with launch-options for push notification tracking

### PR DESCRIPTION
**Pull request by ladanazita**
To add push open tracking, Mixpanel requires that on initialization Mixpanel is launched with options, meaning Segment needs  an instance which includes this `launchOptions` parameter:

```[Mixpanel sharedInstanceWithToken:MIXPANEL_TOKEN launchOptions:launchOptions];```

I will need a ton of feedback on this, as I'm not too familiar with building out this functionality. 

[EPD-307](https://segment.atlassian.net/browse/EPD-307)
[Tracking push notification open rate](https://mixpanel.com/help/questions/articles/how-do-i-track-push-notification-open-rate)

@hankim813 @anoonan

### CODE REVIEW

File: Pod/Classes/SEGMixpanelIntegrationFactory.m [hunk 1]
```diff
@@ -20,11 +20,25 @@ - (instancetype)init
     return self;
 }
 
+
++ (instancetype)instanceWithLaunchOptions: (NSString *)token launchOptions:(NSDictionary *)launchOptions
+{
+    static dispatch_once_t once;
+    static SEGMixpanelIntegrationFactory *sharedInstance;
+    dispatch_once(&once, ^{
+        [Mixpanel sharedInstanceWithToken:token launchOptions:launchOptions];
```
Comment (f2prateek) [inline Pod/Classes/SEGMixpanelIntegrationFactory.m]: Don't initialize this here. Keep the launchOptions as a field in the factory. Then when `createWithSettings` is called, supply the launchOptions to the MixpanelIntegration.

File: Pod/Classes/SEGMixpanelIntegrationFactory.h [hunk 2]
```diff
@@ -1,9 +1,10 @@
 #import <Foundation/Foundation.h>
 #import <Analytics/SEGIntegrationFactory.h>
 
-
 @interface SEGMixpanelIntegrationFactory : NSObject <SEGIntegrationFactory>
-
 + (instancetype)instance;
++ (instancetype)createWithLaunchOptions: (NSString *)token launchOptions:(NSDictionary *)launchOptions;
+@property NSDictionary *launchOptions;
```
Comment (f2prateek) [inline Pod/Classes/SEGMixpanelIntegrationFactory.h]: remove this line.

File: Pod/Classes/SEGMixpanelIntegration.m [hunk 3]
```diff
@@ -23,6 +23,18 @@ - (instancetype)initWithSettings:(NSDictionary *)settings andMixpanel:(Mixpanel
     return self;
 }
 
+-(instancetype)initWithSettings:(NSDictionary *)settings launchOptions:(NSDictionary *)launchOptions
```
Comment (f2prateek) [inline Pod/Classes/SEGMixpanelIntegration.m]: You're not calling this method anywhere. The factory has to supply the launch options to this method for this to work.

**Comment by ladanazita** [comment]
@f2prateek Checks are finally passing!

**Comment by f2prateek** [comment]
LGTM!

**Pull request merged**